sync/handlers/leafs_request_test.go

```diff
@@ -649,6 +651,53 @@ func TestLeafsRequestHandler_OnLeafsRequest(t *testing.T) {
 				assertRangeProofIsValid(t, &request, &leafsResponse, false)
 			},
 		},
+		"reading from snapshot with stale data times out": {
+			prepareTestFn: func() (context.Context, message.LeafsRequest) {
+				// delayedReader allows us to cause the snapshot read context
+				// to timeout, so we can test the trie is read from disk as a fallback.
```

Comment:
Can we make this more obviously correct by just blocking the snapshot on `<-ctx.Done()` ?

This will make it a bit more clear that we wait for the `snapCtx` to timeout instead of needing to think about how many delays occur and what the exact time outs are.

Reply:
Do you find the new version better?
It is a bit tricky since we can't wait for the entire context to expire and need to know how long we should wait, but this makes the math a bit more explicit